File: js/stack&queue/stack.js
const Node = require("../linkedlist/node");

class Stack {
    constructor(){
        this.top = null;
    }

    push(val) {
        let node = new Node(val);
        if(this.top == null)
            this.top = node;
        else {
            node.next = this.top;
            this.top = node;
        }
    }

    pop() {
        if(this.top == null)
            throw new Error("Nothing in the stack");
        else {
            let temp = this.top;
            this.top = this.top.next;
            return temp.value;
        }
    }

    peek = () => this.top == null ? Error("nothing in the stack"): this.top;

    isEmpty = () => this.top == null;
}



module.exports = Stack;

File: csharp/stackNQ/stackNQ/Class/Stack.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace stackNQ.Class
{
    public class Stack<T>
    {
        public Node<T> Top { get; set; }

        public void Push(T val)
        {
            var node = new Node<T>(val);
            if(Top == null)
            {
                Top = node;
            }
            else
            {
                node.Next = Top;
                Top = node;
            }
        }

        public T Pop()
        {
            if (Top == null)
                throw new Exception("There is nothing in the stack");
            else
            {
                var temp = Top;
                Top = Top.Next;
                return temp.Value;
            }
        }

        public T Peek() => Top == null ? throw new Exception("There is nothing in the stack") : Top.Value;

        public bool IsEmpty() => Top == null;
    }
}
